core/adapters/result_collect_test.go

```diff
@@ -0,0 +1,63 @@
+package adapters_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/core/adapters"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+)
+
+func TestResultCollect_Perform(t *testing.T) {
+	var tt = []struct {
+		name                     string
+		json                     string
+		expectedResultCollection []gjson.Result
+		expectedResult           gjson.Result
+	}{
+		{
+			name:                     "empty add bool",
+			json:                     `{"result":false}`,
+			expectedResultCollection: gjson.ParseBytes([]byte(`[false]`)).Array(),
+			expectedResult:           gjson.ParseBytes([]byte(`false`)),
+		},
+		{
+			name:                     "exists add bool",
+			json:                     fmt.Sprintf(`{"result":false,"%s":[false,true]}`, models.ResultCollectionKey),
+			expectedResultCollection: gjson.ParseBytes([]byte(`[false,true,false]`)).Array(),
+			expectedResult:           gjson.ParseBytes([]byte(`false`)),
+		},
+		{
+			name:                     "exists add int",
+			json:                     fmt.Sprintf(`{"result":20,"%s":[false]}`, models.ResultCollectionKey),
+			expectedResultCollection: gjson.ParseBytes([]byte(`[false,20]`)).Array(),
+			expectedResult:           gjson.ParseBytes([]byte(`20`)),
+		},
+		{
+			name:                     "exists add float",
+			json:                     fmt.Sprintf(`{"result":20.214,"%s":[false]}`, models.ResultCollectionKey),
+			expectedResultCollection: gjson.ParseBytes([]byte(`[false,20.214]`)).Array(),
+			expectedResult:           gjson.ParseBytes([]byte(`20.214`)),
+		},
```

Comment:
Can you add a test for what happens when a value exists in `__chainlink_result_collection__` that is not an array? It's an edge case that we are obviously trying to avoid, and I expect it to be fine, but would be good to document what we expect.

Reply:
Interestingly it actually handles that case accurately because gjson's Array function returns [element] if its a single non-array element.